RoomsEyeView/Logic/vtkSlicerRoomsEyeViewModuleLogic.cxx

```diff
@@ -423,13 +443,162 @@ void vtkSlicerRoomsEyeViewModuleLogic::BuildRoomsEyeViewTransformHierarchy()
     vtkErrorMacro("BuildRoomsEyeViewTransformHierarchy: Invalid MRML scene");
     return;
   }
+  vtkSlicerBeamsModuleLogic* beamsLogic = vtkSlicerBeamsModuleLogic::SafeDownCast(this->GetModuleLogic("Beams"));
+  if (!beamsLogic)
+  {
+    vtkErrorMacro("BuildRoomsEyeViewTransformHierarchy: Beams logic cannot be accessed");
+    return;
+  }
 
   // Build IEC hierarchy
   //TODO: Add the REV transform to the IEC transform map and use it for the GetTransform... functions
-  this->IECLogic->BuildIECTransformHierarchy();
+
+  // Create transform nodes if they do not exist
+  for (auto& transformPair : this->IECLogic->GetIecTransforms())
+  {
+    std::string transformNodeName = this->IECLogic->GetTransformNameBetween(transformPair.first, transformPair.second);
+    if (!this->GetMRMLScene()->GetFirstNodeByName(transformNodeName.c_str()))
+    {
+      vtkSmartPointer<vtkMRMLLinearTransformNode> transformNode = vtkSmartPointer<vtkMRMLLinearTransformNode>::New();
+      transformNode->SetName(transformNodeName.c_str());
+      transformNode->SetHideFromEditors(1);
+      std::string singletonTag = std::string("IEC_") + transformNodeName;
+      transformNode->SetSingletonTag(singletonTag.c_str());
+      this->GetMRMLScene()->AddNode(transformNode);
+    }
+  }
+
+  // Organize transform nodes into hierarchy based on IEC Standard 61217 (duplicate IEC logic in MRML).
+  // Set and observe individual transforms from IEC logic to the MRML nodes.

```

Comment:
Explain it here that although we have the concatenated transforms in the IEC logic, we want to define a MRML transform hierarchy because that is how we can set them to the model nodes that are the RT machine parts.